### ISSUE TITLE
Add support for different Dockerfile (Docker 1.5.0)

### ISSUE THREAD

**Issue by birdy-** [docker/fig#956]
Since Docker allows multiple Dockerfile in a single directory, it would be great if docker-compose allows a Dockerfile as param for "build" instead of a directory.

**Comment by k4nar** [comment]
That would be *really* great.

For example let's say I have a Django application, which uses Grunt for asset compilation. Today I have to put one of the container in a separate directory, which doesn't have access to my code at build time. This means that I have to run `npm install` and `grunt install` after the first build.
With the ability to use several Dockerfile in the same directory, this isn't an issue anymore.

tl;dr: several containers might need the code at build-time, hence it is useful.

**Comment by dnephin** [comment]
Duplicate of #834
